build/Targets/Imports.targets

```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" TreatAsLocalProperty="OutDir">
```

Comment:
I'm basing my comments off the scenario that @tmat added to the main PR comment

``` cmd
msbuild Roslyn.sln /p:BaseOutputPath=C:\Temp\Output\ /p:BaseIntermediateOutputPath=C:\Temp\Intermediate\
```

Based on this scenario there is no reason for `OutDir` to be a TreatAsLocalProperty.